Release Notes 99 (#3865)
pulumi/pulumi-hugo: themes/default/content/blog/pulumi-release-notes-99/index.md

meta_image: meta.png
authors:
    - arun-loganathan
    - justin-vanpatten
tags:
    - features
    - release-notes
---

We've had a busy last few months at Pulumi. From shipping a brand new product to Pulumi Cloud, to introducing several major features within Pulumi Cloud and updates to several Pulumi providers - we have them all. Check out [pulumi/pulumi repo](https://github.com/pulumi/pulumi/blob/master/CHANGELOG.md) changelog for CLI enhancements. As always, follow the [new features blogs](/blog/tag/features) to stay updated on the latest Pulumi Cloud features. Let's start with the major updates.

<!--more-->

- [AI](#ai)
  - [Deploy with Pulumi within Pulumi AI](#deploy-with-pulumi-within-pulumi-ai)
  - [Pulumi AI coverage and performance](#pulumi-ai-coverage-and-performance)
- [Pulumi Cloud](#pulumi-cloud)
  - [Pulumi ESC Preview](#pulumi-esc-preview)
  - [Developer Portal Gallery](#developer-portal-gallery)
  - [Deployments GA](#deployments-ga)
  - [Customer Managed Agents](#customer-managed-agents)
  - [MFA](#mfa)
  - [Pricing Calculator](#pricing-calculator)
  - [Historical Views](#historical-views)
- [Core](#core)
- [Providers and Packages](#providers-and-packages)

>   - [AWS 6.0 to AWSX](#aws-60-to-awsx)

Author: ```suggestion
  - [AWSX now uses AWS 6.0](#awsx-now-uses-aws-60)
```

### Pricing Calculator

[Pulumi cost calculator](/pricing/#calculator) for Team Edition enables you to quickly estimate the cost of using Pulumi Cloud. Simply input the number of resources in all your cloud accounts and the percentage of time they will be running to quickly get the total cost per month with a break down of estimated credits, free credits and cost per credit. You can also use the calculator to estimate the cost of using Pulumi Deployments.

![Screenshot of the pricing calculator](../pricing-calculator-blog/pricing_calc.png)

### Historical Views

Stack update page now contains the historical resources in the stack and the historical stack outputs. Simply click on details on each pulumi update to see the resource state and stack output at the time of the update. This history extends all the way back to the first update of the stack. See the [historical views](/blog/update-page-improvements/) blog post for more details.

![Historical views](../update-page-improvements/update-page.png)

## Core

Placeholder

## Providers and Packages

> ### AWS 6.0 to AWSX

Author: ```suggestion
### AWSX now uses AWS 6.0
```

### AWS S3 Express One Zone Support

We [added support](/blog/amazon-s3-express-one-zone/) for [Amazon S3 Express One Zone](https://aws.amazon.com/s3/storage-classes/express-one-zone/), a new storage class offering up to 10X faster performance and reduced request costs by 50%. S3 Express One Zone is ideal for data-intensive applications such as AI/ML, HPC, etc. This update is part of our commitment to provide timely and effective support for the latest AWS offerings.

### New Provider resources

Placeholder

### New Community Providers

Placeholder

## Wrap up

> That's a wrap on our latest Pulumi release notes! With each new feature and improvement, from AI advancements to enhanced cloud deployment tools, we're excited to see how you leverage these updates. Your [feedback](https://github.com/pulumi/pulumi-cloud-requests/issues/new/choose) is crucial in shaping our path forward. tay connected for more updates, and here's to building a more efficient cloud future together!

Author: ```suggestion
That's a wrap on our latest Pulumi release notes! With each new feature and improvement, from AI advancements to enhanced cloud deployment tools, we're excited to see how you leverage these updates. Your feedback is crucial in shaping our path forward, feel free to open issues in our [Pulumi Cloud requests repository](https://github.com/pulumi/pulumi-cloud-requests/issues/new/choose)  or in the [pulumi/pulumi repository](https://github.com/pulumi/pulumi) for CLI feedback. Stay connected for more updates, and here's to building a more efficient cloud future together!
```

  - [New Provider Resources](#new-provider-resources)
  - [New Community Providers](#new-community-providers)
- [Wrap up](#wrap-up)

## AI

### Deploy with Pulumi within Pulumi AI

Pulumi AI's new feature enables [deploying cloud infrastructure using AI prompts](/blog/pulumi-ai-new). Users can rapidly generate and deploy templates through natural language inputs in 'pulumi new' CLI command and new project wizard, significantly streamlining the cloud setup process. Additionally, you now have the option to deploy the Pulumi AI generated Pulumi programs from within [Pulumi AI]([/ai/](https://www.pulumi.com/ai)) and in the New Project Wizard in the Pulumi Cloud console.

{{< video title="Pulumi new AI and deploy" src="../pulumi-ai-new/pulumi-new-ai.mp4" controls="false" autoplay="true" loop="true" >}}

### Pulumi AI coverage and performance

Pulumi AI can now write Pulumi programs for all 150 cloud providers in the [Pulumi registry](/registry/) - up from 20 cloud providers. Pulumi AI has also been updated to OpenAI's GPT-4 Turbo for enhanced performance along with fine tuning of our prompts to improve our AI's ability to write code.

### Pulumi AI Answers

> Pulumi AI Answers has become a go-to repository, hosting an extensive collection of over 50,000 anonymized, infrastructure-as-code commonly asked queries, all carefully selected and presented by Pulumi. As this repository continues to expand, we're seeing an exponential increase in user traffic, underscoring the immense value it offers. It's inspiring to see the Pulumi community actively leveraging it as they build and scale their cloud infrastructure

Author: ```suggestion
[Pulumi AI Answers](https://www.pulumi.com/ai/answers), launch late last year, has become a go-to repository, hosting an extensive collection of over 50,000 anonymized, infrastructure-as-code commonly asked queries, all carefully selected and presented by Pulumi. As this repository continues to expand, we're seeing an exponential increase in user traffic, underscoring the immense value it offers. It's inspiring to see the Pulumi community actively leveraging it as they build and scale their cloud infrastructure
```
